include/oboe/AudioStreamBase.h

```diff
@@ -195,6 +195,28 @@ class AudioStreamBase {
     bool                            mFormatConversionAllowed = false;
     // Control whether and how Oboe can convert sample rates to achieve optimal results.
     SampleRateConversionQuality     mSampleRateConversionQuality = SampleRateConversionQuality::None;
+
+    /** Validate the existing stream configurations */
+    virtual Result isValidConfig() {
+        if (mFormat < AudioFormat::Invalid || mFormat > AudioFormat::Float) {
+            return Result::ErrorInvalidFormat;
+        }
+        if ((mSharingMode != SharingMode::Exclusive && mSharingMode != SharingMode::Shared) ||
```

Comment:
This is extremely difficult to parse. Please consider splitting into individual methods e.g. `isSharingModeValid()`, `isPerformanceModeValid()` etc to make it easier to read.

Reply:
Additionally, it may be worth logging an error inside each method so the caller knows exactly what was wrong.